### PR TITLE
Simplify svg font expansion logic.

### PR DESCRIPTION
**Pull request by anntzer**
We can delay quoting and deduplication until quite late.

## PR Checklist

**Tests and Styling**
- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

### CODE REVIEW

File: lib/matplotlib/backends/backend_svg.py [hunk 1]
```diff
@@ -1154,47 +1154,31 @@ def _draw_text_as_text(self, gc, x, y, s, prop, angle, ismath, mtext=None):
...
+                    # (we accept 3 ways CSS only supports 1)
+                    for name in fm.FontManager._expand_aliases(fn):
+                        yield _normalize_sans(name)
+                # whether a generic name or a family name, it must appear at
+                # least once
+                yield fn
+
+            def _get_all_quoted_names(prop):
+                # only quote specific names, not generic names
+                return [name if name in fm.font_family_aliases else repr(name)
```
Comment (timhoffm) [inline lib/matplotlib/backends/backend_svg.py:1174]: We have used `repr()` before and it seems to have worked, but technically `repr()` is not exactly quoting. Would both single and double quotes be ok? Unlikely, but in theory Python could change that in repr (or possibly more likely an interpreter other than CPython could use other quotes). Also, can we be sure we will never have byte strings here?

It's more defensive to f-string the quotes explicitly.

**Review by timhoffm — APPROVED** [review]
Optional comment on quoting.

File: lib/matplotlib/backends/backend_svg.py (hunk 1 above)
Comment (tacaswell) [reply lib/matplotlib/backends/backend_svg.py:1174]: Do browsers / svg viewers / svg care about double vs single quotes?

Comment (tacaswell) [reply lib/matplotlib/backends/backend_svg.py:1174]: I think we should be safe on byte strings though (I would say a byte string as a font name is a bug or at least un-defined behavior).

**Review by tacaswell — APPROVED** [review]

**Comment by tacaswell** [comment]
Both test failures are

```
 E       PIL.UnidentifiedImageError: cannot identify image file '/home/vsts/work/1/s/result_images/test_axes/hexbin_empty.png'
```

which is because we now have to tests that use the same underlying file

https://github.com/matplotlib/matplotlib/blob/7de767e927b4b4f5212e268c6432107116752d85/lib/matplotlib/tests/test_axes.py#L897-L909

File: lib/matplotlib/backends/backend_svg.py (hunk 1 above)
Comment (anntzer) [reply lib/matplotlib/backends/backend_svg.py:1174]: Single and double quotes are equally valid, and backslash-escaping quotes is valid as well: https://developer.mozilla.org/en-US/docs/Web/CSS/string

**Comment by tacaswell** [comment]
Merged and left the discussion about quotes to #24080

**Pull request merged**